src/com/xilinx/rapidwright/examples/CounterGenerator.java

```diff
@@ -0,0 +1,313 @@
+            accepts(COUNT_DOWN_OPT).withOptionalArg().ofType(Boolean.class).defaultsTo(countDown).describedAs("Counts down instead of up");
+            accepts(STEP_OPT).withOptionalArg().ofType(Long.class).defaultsTo(step).describedAs("The amount to increment/decrement each clock cycle");
+            accepts(INIT_OPT).withOptionalArg().ofType(Long.class).defaultsTo(initValue).describedAs("The value the counter starts at");
+            accepts(VERBOSE_OPT).withOptionalArg().ofType(Boolean.class).defaultsTo(verbose).describedAs("Print verbose output");
+            acceptsAll( Arrays.asList(HELP_OPT, "?"), "Print Help" ).forHelp();
+        }};
+
+        return p;
+    }
+
+    private static void printHelp(OptionParser p) {
+        MessageGenerator.printHeader("Counter Generator");
+        System.out.println("This RapidWright program creates a placed and routed DCP for UltraScale+ devices that contains \n"
+                + "imported into UltraScale+ designs to perform addition or subtraction.  See \n"
+                + "RapidWright documentation for more information.\n");
+        try {
+            p.accepts(OUT_DCP_OPT).withOptionalArg().defaultsTo("counter.dcp").describedAs("Output DCP File Name");
+            p.printHelpOn(System.out);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Implements a parameterizable counter for an UltraScale/UltraScale+ device and routes the counter using RWRoute.
+     */
+    public static void main(String[] args) {
+        // Extract program options
+        OptionParser p = createOptionParser();
+        OptionSet opts = p.parse(args);
+        boolean verbose = (boolean) opts.valueOf(VERBOSE_OPT);
+        if (opts.has(HELP_OPT)) {
+            printHelp(p);
+            return;
+        }
+
+        CodePerfTracker t = verbose ? new CodePerfTracker(AddSubGenerator.class.getSimpleName(),true).start("Init") : null;
+
+        String partName = (String) opts.valueOf(PART_OPT);
+        String designName = (String) opts.valueOf(DESIGN_NAME_OPT);
+        String outputDCPFileName = (String) opts.valueOf(OUT_DCP_OPT);
+        double clkPeriodConstraint = (double) opts.valueOf(CLK_CONSTRAINT_OPT);
+        int width = (int) opts.valueOf(WIDTH_OPT);
+        String sliceName = (String) opts.valueOf(SLICE_SITES_OPT);
+        boolean countDown = (boolean) opts.valueOf(COUNT_DOWN_OPT);
+        long step = (long) opts.valueOf(STEP_OPT);
+        long initValue = (long) opts.valueOf(INIT_OPT);
+
+        // Perform some error checking on inputs
+        Part part = PartNameTools.getPart(partName);
+        if (part == null || part.isSeries7() || part.isVersal()) {
+            throw new RuntimeException("ERROR: Invalid/unsupported part " + partName + ".");
+        }
+
+        if (width <= 0) {
+            throw new RuntimeException("ERROR: The counter's width must be greater than 0.");
```

Comment:
When I load a counter generated with a width of 1 into Vivado, I get the following error:
![image](https://github.com/Xilinx/RapidWright/assets/20803522/8eb02d8e-b308-4097-885b-c005fb249298)

Perhaps you just want to support counters of width greater than 1 -- if someone needed a counter of width 1, this class might be overkill.